linera-core/src/client/mod.rs

```diff
@@ -2337,6 +2337,7 @@ impl<Env: Environment> ChainClient<Env> {
 
         let mutex = self.state().client_mutex();
         let _guard = mutex.lock_owned().await;
+        // TOOD: We shouldn't need to call this explicitly.
```

Comment:
This was to guard against the case where this is called even if there already is a pending block in progress. Not sure if it's still necessary.